components/director/internal/tenantfetcher/event_page.go

```diff
@@ -95,41 +111,86 @@ func (ep eventsPage) eventDataToTenant(eventType EventsType, eventData []byte) (
 	if !gjson.Valid(jsonPayload) {
 		return nil, errors.Errorf("invalid json payload")
 	}
-	if eventType == CreatedEventsType && ep.fieldMapping.DiscriminatorField != "" {
+	if eventType == CreatedAccountType && ep.fieldMapping.DiscriminatorField != "" {
 		discriminatorResult := gjson.Get(jsonPayload, ep.fieldMapping.DiscriminatorField)
 		if !discriminatorResult.Exists() {
-			return nil, errors.Errorf("invalid format of %s field", ep.fieldMapping.DiscriminatorField)
+			return nil, invalidFieldFormatError(ep.fieldMapping.DiscriminatorField)
 		}
 		if discriminatorResult.String() != ep.fieldMapping.DiscriminatorValue {
 			return nil, nil
 		}
 	}
-	idResult := gjson.Get(jsonPayload, ep.fieldMapping.IDField)
-	if !idResult.Exists() {
-		return nil, errors.Errorf("invalid format of %s field", ep.fieldMapping.IDField)
+
+	id, err := determineTenantID(jsonPayload, ep.fieldMapping)
+	if err != nil {
+		return nil, err
 	}
 
 	nameResult := gjson.Get(jsonPayload, ep.fieldMapping.NameField)
 	if !nameResult.Exists() {
-		return nil, errors.Errorf("invalid format of %s field", ep.fieldMapping.NameField)
-	}
-
-	customerIDResult := gjson.Get(jsonPayload, ep.fieldMapping.CustomerIDField)
-	if !customerIDResult.Exists() {
-		log.D().Warnf("Missig or invalid format of field: %s for tenant with ID: %s", ep.fieldMapping.CustomerIDField, idResult.String())
+		return nil, invalidFieldFormatError(ep.fieldMapping.NameField)
 	}
 
 	subdomain := gjson.Get(jsonPayload, ep.fieldMapping.SubdomainField)
 	if !subdomain.Exists() {
-		return nil, errors.Errorf("invalid format of %s field", ep.fieldMapping.SubdomainField)
+		log.D().Warnf("Missig or invalid format of field: %s for tenant with ID: %s", ep.fieldMapping.SubdomainField, id)
+	}
+
+	entityType := gjson.Get(jsonPayload, ep.fieldMapping.EntityTypeField)
+	if !entityType.Exists() {
+		return nil, invalidFieldFormatError(ep.fieldMapping.EntityTypeField)
+	}
+
+	region := ""
+	parentID := ""
+	tenantType := tenant.TypeToStr(tenant.Account)
+	globalAccountRegex := regexp.MustCompile("^GLOBALACCOUNT_.*|GlobalAccount")
+	if globalAccountRegex.MatchString(entityType.String()) {
+		customerIDResult := gjson.Get(jsonPayload, ep.fieldMapping.CustomerIDField)
+		if !customerIDResult.Exists() {
+			log.D().Warnf("Missig or invalid format of field: %s for tenant with ID: %s", ep.fieldMapping.CustomerIDField, id)
+		} else {
+			parentID = customerIDResult.String()
+		}
+	} else {
+		regionField := gjson.Get(jsonPayload, ep.fieldMapping.RegionField)
+		if !regionField.Exists() {
+			return nil, invalidFieldFormatError(ep.fieldMapping.RegionField)
+		}
+		region = regionField.String()
+		parentIDField := gjson.Get(jsonPayload, ep.fieldMapping.ParentIDField)
+		if !parentIDField.Exists() {
+			return nil, invalidFieldFormatError(ep.fieldMapping.ParentIDField)
+		}
+		parentID = parentIDField.String()
+		tenantType = tenant.TypeToStr(tenant.Subaccount)
```

Comment:
This one should be configurable - `globalAccountRegex := regexp.MustCompile("^GLOBALACCOUNT_.*|GlobalAccount")`

Also this whole block is a bit larger than we would usually want. My suggestion is to extract two functions here, which can be responsible for building the tenant of the type we want.